### DISCUSSION TITLE
Comment function for Achievement Code

### DISCUSSION THREAD

**Discussion by ghost** [RetroAchievements/RAWeb#782] [Ideas]
As mentioned in our #raweb Discord Channel already, adding a Comment function to Achievement Code would help us analyse Code easier and faster as the Toolkit is constantly getting updated with new features and the Achievement Code gets more complex over the time.

@meleu also mentioned to add the Comment function into the Toolkit so Devs can add Comments while creating Achievements.

**Comment by Jamiras** [comment]
see also #155

**Comment by meleu** [comment]
The proposal here would be just add a TEXT type field on the Achievements DB table.

This way each achievement can have a comment and devs can use it to better describe the achievement's logic.

If I understood correctly, #155 propose adding a comment field for each requirement, which would be a bit overkill. A comment for the whole achievement is a lot simpler/easier to implement and enough to give a good explanation of the achievement's logic.

**Comment by televandalist** [comment]
I'd love this. Some logic needs to be a bit off-kilter for an achievement to work properly and this would definitely help. Would also help as a learning tool for jr-devs.

**Comment by kdecks** [comment]
I would use it. Seems it would be good so long as it's not stored in the achievement file

**Comment by ghost** [comment]
The Trigger information already exist in form of the measuring feature which has been implemented recently by @jamiras.

I agree with @kdecks, Comments for Achievement saved in files isn't necessary, they should be just saved on the Site or at least excluded from xxx-user.txt or else its gets to spammy.
